packages/blockly/docs/authoring.md

```diff
@@ -0,0 +1,160 @@
+
+Different blocks generate code in different ways. Some automatically produce code, and using these blocks will require you to set up your simulation code in specific ways. Other blocks require authors to provide their own code.
+
+### Starter Blocks
+
+#### Setup
+
+This block defines the `setup` function, which will include the code generated by all blocks contained in the setup block. In order for this block to do anything, include `setup();` at the bottom of the simulation code.
+
+#### Go
+
+This block defines the `sim.afterTick` method. If you define your own `sim.afterTick` method in the simulation code, the method defined by this block will override it.
+
+#### On Mouse Click
+
+This block currently defines a function called `onClick`, which can be used as an event listener (see [Atomic Agents Vis Interaction](https://gjmcn.github.io/atomic-agents-vis/#/?id=interaction)).
+
+### Creator Blocks
+
+These custom blocks are used to create actors of different types in the simulation. They generate code like:
+
+`create_[type]([count], [callback]);`
+
+- `type` is the Object Name specified in the edit block UI
+- `count` is a value specified by a user
+- `callback` is a function like `(agent) => { [statements] }`
+  - `agent` is the newly created agent (the callback should be called with every agent created)
+  - `statements` is the code generated by blocks contained within the creator block (usually setters)
+  - If the creator block contains no blocks, the callback will NOT be included
+
+To use a creator block, the author needs to define a function called `create_[type]` for each type the creator block allows.
+- The function should take a `count` and a `callback` (which might be `undefined`).
+- It should create `count` actors.
+- Each actor should have a label `type` applied to it.
+  - Note that the label should be exactly the same as the `type`--be mindful of singular vs plural here.
+- If provided, `callback` should be called on each new actor.
+
+#### Example
+
+This example is from `predator-prey`. The first function, `create_a_wolf`, creates a single new actor. Note that "wolves" is applied as a label. This function is called by `create_wolves`, which is what is directly called by the code generated by the creator block. This function calls `ceate_a_wolf` `count` times, calling `callback` on each newly created agent (assuming `callback` is actually defined).
+
+```
+function create_a_wolf(props) {
+  const { color, energy, x, y } = props ?? {};
+  const agent = new AA.Actor();
+  agent.radius = 10;
+  agent.vel = AA.Vector.randomAngle(1.5);
+  agent.vis({ image: wolfImage, tint: color ?? "0x333333" });
+  agent.label("wolves", true);
+  agent.state = { energy: energy ?? wolfEnergy };
+  agent.x = x ?? Math.random() * sim.width;
+  agent.y = y ?? Math.random() * sim.height;
+
+  agent.addTo(sim);
+  return agent;
+};
+function create_wolves(num, callback) {
```

Comment:
Nitpick, sorry, but I think a line break between the two functions would improve legibility.